ga4gh/datamodel/variants.py

```diff
@@ -547,3 +596,162 @@ def buildMetadata(
                         number="{}".format(value.number),
                         description=description))
         return ret
+
+
+class HtslibVariantAnnotationSet(HtslibVariantSet):
+    """
+    Class representing a single variant annotation set backed by a directory of
+    indexed VCF or BCF files.
+    """
+    def __init__(self, parentContainer, localId, dataDir, backend):
+        super(HtslibVariantAnnotationSet, self).__init__(
+            parentContainer, localId, dataDir, backend)
+        self.compoundIdClass = datamodel.VariantAnnotationSetCompoundId
+        self._compoundId = datamodel.VariantAnnotationSetCompoundId(
+            self.getCompoundId(), 'variantannotations')
+        self._sequenceOntology = backend.getOntologies() \
+            .get('sequence_ontology')
+
+    def getVariantAnnotations(self, referenceName, startPosition, endPosition):
+        """
+        Returns an iterator over the specified variant annotaitons.
+        """
+        if referenceName in self._chromFileMap:
+            varFileName = self._chromFileMap[referenceName]
+            referenceName, startPosition, endPosition = \
+                self.sanitizeVariantFileFetch(
+                    referenceName, startPosition, endPosition)
+            cursor = self.getFileHandle(varFileName).fetch(
+                referenceName, startPosition, endPosition)
+            for record in cursor:
+                yield self.convertVariantAnnotation(record)
+
+    def convertLocation(self, pos):
+        if pos == '':
```

Comment:
I worry a bit about comparing equality with the empty string. There's a potential Python2/Python3 issue here with bytes/str.  Perhaps ``if len(pos) == 0``? I'm not sure what the right thing to do is here, so feel free to ignore this point.